modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/spring/ByteBuddySpringCachesManager.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.spring;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import net.bytebuddy.asm.Advice;
+import org.slf4j.Logger;
+import org.springframework.core.SpringVersion;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Optional.empty;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * ByteBuddy aspect implementation that stores Soring {@link org.springframework.util.ConcurrentReferenceHashMap} caches, offering
+ * a cleanup method. The use the cleanup feature, this aspect must be instrumented through
+ * {@link ByteBuddySpringCacheInstrumentator#instrumentForCleanup(ClassLoader)} call.
+ * 
+ * @see ByteBuddySpringCacheInstrumentator#instrumentForCleanup(ClassLoader)
+ */
+public class ByteBuddySpringCachesManager {
+
+  private static final Logger LOGGER = getLogger(ByteBuddySpringCachesManager.class);
+  public static final Cache<Object, Optional<?>> springCaches = Caffeine.newBuilder().weakKeys().build();
```

Comment:
Using weakKeys in this cache could make the clearCaches to "forget" to clean some cache. Wouldn't a plain old ConcurrentHashMap work?